bson/__init__.py

```diff
@@ -517,22 +516,56 @@ def _get_decimal128(
 }
 
 
+class LazyValue:
+    """A BSON sub-document or array that can be lazily loaded."""
+
+    def __init__(self, view, is_array, codec_options):
+        self.__inflated = None
+        self.__view = view
+        self.__is_array = is_array
+        self.__codec_options = codec_options
+
+    @property
+    def raw(self):
+        return memoryview(self.__view)
+
+    @property
+    def value(self):
+        if not self.__inflated:
+            self.__inflated = decode(self.__view, self.__codec_options)
+            if self.__is_array:
+                assert self.__inflated is not None
+                self.__inflated = [a for a in self.__inflated.values()]
```

Comment:
Actually, we pass a flag through to treat an array as a stream of documents and do the transformation on the spot.  Then we don't need a LazyValue at all.  Does that sound reasonable?

Reply:
How would we be able to tell the difference between an array and a bytes?

Reply:
This would only be used by `OpMsg.raw_response`, to explicitly tell `_get_array` (and the C ext equivalent) to decode differently for a given field name.

Reply:
Oh right. Yeah I think that would work nicely. Then we wouldn't need a LazyValue or _cbson_array_of_documents_to_buffer.